Shopify/mobile-buy-sdk-ios
# Bugfix/314 test errors

Various changes to attempt to make tests faster, easier to configure, and more reliable.

Some things this PR does:

- Private data has been moved to environment variables. These can be set in a non-shared scheme, or in the CI environment, without committing to a repo.
- Configuration file has been standardized regardless whether we are using mocked responses or live server connection.
- Separate the handling of shop storefront data from configuration data:
  - break out config settings to be handled specially (domain, app id, api key, merchant id)
  - config settings can be provided in env variables (settable in scheme in Xcode)
  - config settings can alternately be provided in a file, for use in CI using same generation trick
  - the file is named `test_shop_config.json`, and exists and should be (and stay) empty
    - in any case, it is no longer tracked in version control
  - (the inability to pass env vars to `xcodebuild` to pass on to test harness is infuriating)
- Some constants have been standardized, converted from preprocessor macros.
- Removes some empty completion blocks that weren't needed
- The email address used to test sign up failure was used to ensure we get two errors from the server.

The BuildKite configuration was changed for this. There are, temporarily (until this PR is merged or rejected) two buildkite build steps: the original, run only for master and develop, and a new one run only for this branch.

## Testing Steps
- test this locally the usual way -- should work with mock data
- test this locally using environment variables:
  - clone `Mobile Buy SDK Tests` scheme
  - rename it something like `Mobile Buy SDK Tests - production`
  - make it non-shared, so it isn't saved into git
  - enter environment variables for the Test action, as follows:

![clone scheme](https://cloud.githubusercontent.com/assets/186680/18366224/d614052e-75e3-11e6-922d-33fabeaad877.png)

  - run tests with the new `production` scheme selected

Still getting a failure `-testCancellationDuringPolling`, however. 😢 

Fixes #314 
Fixes #336 

@davidmuzi @gabrieloc @dbart01
